openedx/core/lib/edx_api_utils.py

```diff
@@ -45,8 +48,11 @@ def get_edx_api_data(api_config, user, resource, resource_id=None, querystring=N
             return cached
 
     try:
-        jwt = get_id_token(user, api_config.OAUTH2_CLIENT_NAME)
-        api = EdxRestApiClient(api_config.internal_api_url, jwt=jwt)
+        if api_client_name == 'commerce':
+            api = ecommerce_api_client(user)
```

Comment:
@ahsan-ul-haq this isn't quite what I had in mind. It's a little too rigid. Instead of forcing callers to pass in a string and using that in a conditional, allow callers to pass in a ready-made API client (e.g., `ecommerce_api_client(user)`). If none is provided, default to the behavior that was already here.